fwoa-tools/src/migrationUtils.test.ts

```diff
@@ -0,0 +1,156 @@
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { createWriteStream, WriteStream } from 'fs';
+import AWS from 'aws-sdk';
+import { InitiateAuthRequest } from 'aws-sdk/clients/cognitoidentityserviceprovider';
+import * as AWSMock from 'aws-sdk-mock';
+import { checkConfiguration, getFhirClient, getFhirClientSMART } from './migrationUtils';
+
+const env = process.env;
+AWSMock.setSDKInstance(AWS);
+
+describe('MigrationUtils', () => {
+  beforeEach(() => {
+    expect.hasAssertions();
+    AWSMock.restore();
+  });
+
+  afterEach(() => {
+    AWSMock.restore();
+    jest.resetModules();
+    process.env = env;
+  });
+  describe('getFhirClient', () => {
+    test('throws error when environment variable is missing', async () => {
+      await expect(getFhirClient()).rejects.toThrowError('API_URL environment variable is not defined');
+      process.env.API_URL = 'https://fake-api-url.com/dev';
+      await expect(getFhirClient()).rejects.toThrowError('API_KEY environment variable is not defined');
+      process.env.API_KEY = 'fake-api-key';
+      await expect(getFhirClient()).rejects.toThrowError(
+        'API_AWS_REGION environment variable is not defined'
+      );
+      process.env.API_AWS_REGION = 'us-east-1';
+      await expect(getFhirClient()).rejects.toThrowError(
+        'COGNITO_CLIENT_ID environment variable is not defined'
+      );
+      process.env.COGNITO_CLIENT_ID = 'fake-cognito-client-id';
+      await expect(getFhirClient()).rejects.toThrowError(
+        'COGNITO_USERNAME environment variable is not defined'
+      );
+    });
+
+    test('should call Cognito with correct parameters', async () => {
+      process.env.API_URL = 'https://fake-api-url.com/dev';
+      process.env.API_KEY = 'fake-api-key';
+      process.env.API_AWS_REGION = 'us-east-1';
+      process.env.COGNITO_CLIENT_ID = 'fakeCognitoClientId';
+      process.env.COGNITO_USERNAME = 'fake-cognito-user-name';
+      process.env.COGNITO_PASSWORD = 'fake-cognito-password';
+
+      AWSMock.mock(
+        'CognitoIdentityServiceProvider',
+        'initiateAuth',
+        //eslint-disable-next-line @typescript-eslint/ban-types
+        (params: InitiateAuthRequest, callback: Function) => {
+          expect(params).toMatchInlineSnapshot(`
+            Object {
+              "AuthFlow": "USER_PASSWORD_AUTH",
+              "AuthParameters": Object {
+                "PASSWORD": "fake-cognito-password",
+                "USERNAME": "fake-cognito-user-name",
+              },
+              "ClientId": "fakeCognitoClientId",
+            }
+          `);
+          callback(null, { AuthenticationResult: { IdToken: 'take-id-token' } });
+        }
+      );
+      const fhirClient = await getFhirClient();
+      expect(fhirClient.defaults.baseURL).toEqual('https://fake-api-url.com/dev');
+      expect(fhirClient.defaults.headers).toEqual(
+        expect.objectContaining({ Authorization: 'Bearer take-id-token' })
```

Comment:
nit: typo fake. We may want to consult the runbook for appropriate testing variable names to prevent the code scanning alerts?